include/intersection_point.h

```diff
@@ -9,10 +9,12 @@ class IntersectionPoint {
   Vertex position_;
```

Comment:
No??? We still need intersection point position when we calculate direction to light source? 
Or should we do ray.position + ray.direction*z every time :O ?